dcpy/utils/versions.py

```diff
@@ -177,6 +177,23 @@ def parse(v: str) -> Version:
             )
 
 
+def sort(versions: list[Version]) -> list[Version]:
+    version_types = set([type(v) for v in versions])
+    # can't compare different types
+    if len(version_types) != 1:
+        # only handle date-like versions
+        if version_types.issubset(set([Date, FirstOfMonth])):
+            return sorted(
+                versions,
+                key=lambda version: datetime.strptime(version.label, "%y-%m-%d"),
```

Comment:
Not sure how tests are passing - this line throws an error when I try to run anything locally. %y should b %Y. @HengJiang0206 if you fix this in your branch, I think things should start working. Sorry for the mysterious error messages

Reply:
Ah - versions in the real world are %Y, while in the test cases we have %y (24 instead of 2024, etc)

Reply:
I guess we should have just stored a `date` object within the `Date` version

Reply:
@fvankrieken It worked! Thanks !

Reply:
@fvankrieken would love to store a `date` object in these `Version` dataclasses